ingestion-edge/pubsub_emulator.py

```diff
@@ -56,23 +56,29 @@ def create_server(self):
         pubsub_pb2_grpc.add_SubscriberServicer_to_server(self, self.server)
         self.server.start()
 
-    def CreateTopic(self, request, context):  # noqa: D403
+    def CreateTopic(
+        self, request: pubsub_pb2.Topic, context: grpc.ServicerContext
+    ):  # noqa: D403
         """CreateTopic implementation."""
         if request.name in self.topics:
             context.set_code(grpc.StatusCode.ALREADY_EXISTS)
         else:
             self.topics[request.name] = set()
         return request
 
-    def DeleteTopic(self, request, context):  # noqa: D403
+    def DeleteTopic(
+        self, request: pubsub_pb2.DeleteTopicRequest, context: grpc.ServicerContext
+    ):  # noqa: D403
         """DeleteTopic implementation."""
         try:
             self.topics.pop(request.topic)
         except KeyError:
             context.set_code(grpc.StatusCode.NOT_FOUND)
         return empty_pb2.Empty()
 
-    def CreateSubscription(self, request, context):  # noqa: D403
+    def CreateSubscription(
+        self, request: pubsub_pb2.Subscription, context: grpc.ServicerContext
+    ):  # noqa: D403
```

Comment:
You can probably get rid of these D403 comments by wrapping CreateSubscription in double backticks to mark it as inline code in ReStructured Text syntax.